analyzers/src/SonarAnalyzer.CFG/ShimLayer/LightupHelpers.cs

```diff
@@ -478,6 +478,200 @@ static bool FallbackAccessor(TSyntax syntax, TKey key, out TValue value)
+        {
+            throw new InvalidOperationException();
+        }
+
+        if (!typeof(TThird).GetTypeInfo().IsAssignableFrom(thirdType.GetTypeInfo()))
+        {
+            throw new InvalidOperationException();
+        }
+
+        var methods = type.GetTypeInfo().GetDeclaredMethods(methodName);
+        MethodInfo method = null;
+        foreach (var candidate in methods)
+        {
+            var parameters = candidate.GetParameters();
+            if (parameters.Length != 4)
+            {
+                continue;
+            }
+
+            if (Equals(firstType, parameters[0].ParameterType)
+                && Equals(secondType, parameters[1].ParameterType)
+                && Equals(thirdType, parameters[2].ParameterType)
+                && Equals(typeof(TValue).MakeByRefType(), parameters[3].ParameterType))
+            {
+                method = candidate;
+                break;
+            }
+        }
+
+        if (method == null)
+        {
+            return FallbackAccessor;
+        }
+
+        if (method.ReturnType != typeof(bool))
+        {
+            throw new InvalidOperationException();
+        }
+
+        var senderParameter = Expression.Parameter(typeof(TSender), "sender");
+        var firstParameter = Expression.Parameter(typeof(TFirst), "first");
+        var secondParameter = Expression.Parameter(typeof(TSecond), "second");
+        var thirdParameter = Expression.Parameter(typeof(TThird), "third");
+        var valueParameter = Expression.Parameter(typeof(TValue).MakeByRefType(), "value");
+        Expression instance =
+            type.GetTypeInfo().IsAssignableFrom(typeof(TSender).GetTypeInfo())
+            ? (Expression)senderParameter
+            : Expression.Convert(senderParameter, type);
+        Expression first =
+            firstType.GetTypeInfo().IsAssignableFrom(typeof(TFirst).GetTypeInfo())
+            ? (Expression)firstParameter
+            : Expression.Convert(firstParameter, firstType);
+        Expression second =
+            secondType.GetTypeInfo().IsAssignableFrom(typeof(TSecond).GetTypeInfo())
+            ? (Expression)secondParameter
+            : Expression.Convert(secondParameter, secondType);
+        Expression third =
+            thirdType.GetTypeInfo().IsAssignableFrom(typeof(TThird).GetTypeInfo())
+            ? (Expression)thirdParameter
+            : Expression.Convert(thirdParameter, thirdType);
+
+        Expression<TryGetValueAccessor<TSender, TFirst, TSecond, TThird, TValue>> expression =
+            Expression.Lambda<TryGetValueAccessor<TSender, TFirst, TSecond, TThird, TValue>>(
+                Expression.Call(instance, method, first, second, third, valueParameter),
+                senderParameter,
+                firstParameter,
+                secondParameter,
+                thirdParameter,
+                valueParameter);
+        return expression.Compile();
+    }
```

Comment:
Would it be worth refactoring so the three overloads of `CreateTryGetValueAccessor` reuse parts of their code?